### PR TITLE
Add some nvchannel ioctls

### PR DESCRIPTION
**Pull request by averne**

### CODE REVIEW

**Comment by fincs** [comment]
Interesting additions. What are these used for?

**Comment by averne** [comment]
Jpeg decoding using nvjpg, you can see my linux code [here](https://github.com/averne/oss-nvjpg), there's still some work that needs to be done on the hos side though.

File: nx/include/switch/nvidia/ioctl.h [hunk 1]
```diff
@@ -247,3 +274,9 @@ Result nvioctlChannel_SetPriority(u32 fd, u32 priority);
 Result nvioctlChannel_SetTimeout(u32 fd, u32 timeout);
 Result nvioctlChannel_AllocGpfifoEx2(u32 fd, u32 num_entries, u32 flags, u32 unk0, u32 unk1, u32 unk2, u32 unk3, nvioctl_fence *fence_out);
 Result nvioctlChannel_SetUserData(u32 fd, void* addr);
+Result nvioctlChannel_Submit(u32 fd, nvioctl_cmdbuf *cmdbufs, u32 num_cmdbufs, nvioctl_reloc *relocs, nvioctl_reloc_shift *reloc_shifts, u32 num_relocs,
```
Comment (fincs) [inline nx/include/switch/nvidia/ioctl.h:277]: Everything except for `fences` should be pointers to const structs, for const correctness reasons.

File: nx/include/switch/nvidia/ioctl.h [hunk 2]
```diff
@@ -247,3 +274,9 @@ Result nvioctlChannel_SetPriority(u32 fd, u32 priority);
...
+    nvioctl_syncpt_incr *syncpt_incrs, u32 num_syncpt_incrs, nvioctl_fence *fences, u32 num_fences);
+Result nvioctlChannel_GetSyncpt(u32 fd, u32 module_id, u32 *syncpt);
+Result nvioctlChannel_GetModuleClockRate(u32 fd, u32 module_id, u32 *freq);
+Result nvioctlChannel_MapCommandBuffer(u32 fd, nvioctl_command_buffer_map *maps, u32 num_maps, bool compressed);
```
Comment (fincs) [inline nx/include/switch/nvidia/ioctl.h:281]: Same here

File: nx/include/switch/nvidia/ioctl.h [hunk 3]
```diff
@@ -247,3 +274,9 @@ Result nvioctlChannel_SetPriority(u32 fd, u32 priority);
...
+Result nvioctlChannel_UnmapCommandBuffer(u32 fd, nvioctl_command_buffer_map *maps, u32 num_maps, bool compressed);
```
Comment (fincs) [inline nx/include/switch/nvidia/ioctl.h:282]: Same here

File: nx/include/switch/nvidia/ioctl.h (hunk 2 above)
Comment (averne) [reply nx/include/switch/nvidia/ioctl.h:281]: Those can't be const because the driver will write the iova to the structs, same for unmap.

Comment (fincs) [reply nx/include/switch/nvidia/ioctl.h:281]: Ah, missed the inout; thanks.

However, in Unmap the copyback seems to be missing.

Comment (averne) [reply nx/include/switch/nvidia/ioctl.h:281]: The driver doesn't seem to clear the iova after unmapping, maybe this can be made const in that case?

Comment (fincs) [reply nx/include/switch/nvidia/ioctl.h:281]: Sure.

**Review by fincs — APPROVED** [review]

**Review by yellows8 — APPROVED** [review]

**Pull request merged**